### DISCUSSION TITLE
FluentLayout and MainLayout.razor

### DISCUSSION THREAD

**Discussion by RoySalisbury** [microsoft/fluentui-blazor#990] [General]
I think this is a .NET 8 Blazor issue with the rendermode, but I would expect it to be a more prominent "issue".

Out of the box, using basic the fluentui template, the NavMenu does not collapse when the icon is clicked.  I would have expected that the basic out of the box sample would either work, or not include the functionality.  It makes a library look cheap when it actually shows off some functionality that does not work.

There has to be a "workaround" for this exact case (even if its not the "blazor way").  This type of thing was being implemented way before Blazor, so its possible.  

At the very least, it should be noted in a very prominent way that controls placed in the MainLayout can not be interactive (and explain why).

NOTE: I know you can change the Route @rendermode="InteractiveServer".  What I am saying is there should be a simplier "fallback" if you don't want the entire site under this mode.

**Comment by vnbaaij** [comment]
We will release an update to the Templates soon (v4.0.1) wich will change the behaviors as follows:

| Rendermode / Interactivity | NavMenu @rendermode | NavMenu Collapsible | WC script in body |
|----------------------------|---------------------|---------------------|-------------------|
| SSR / not applicable       | ❌                   | ❌                   | ✅                 |
| Server / Global            | ❌                   | ✅                   | ❌                 |
| Server / Per Page          | ✅                   | ✅                   | ❌                 |
| WebAssembly / Global*      | ❌                   | ✅                   | ❌                 |
| WebAssembly / Per Page*    | ✅                   | ✅                   | ❌                 |
| Auto / Global*             | ❌                   | ✅                   | ❌                 |
| Auto / Per Page*           | ✅                   | ✅                   | ❌                 |

*NavMenu moved to Client project

Yes, you could do this collapsing in a non-Blazor CSS only way, but this is a means to show how to do it **with** our component and Blazor.

**Reply by RoySalisbury** [reply]
I think this will be very helpful.
